biodiverskripsi.R

```diff
@@ -195,10 +196,146 @@ univ_count
+# Check for anything ended with sp., sp ., Sp., and spp.
+grep("^[A-Za-z]+\\s{1,2}[Ss]?p{1,2}\\s?\\.$", merged_data$scientificName, perl=TRUE, value=TRUE)
+# Remove sp., sp ., Sp., and spp.
+merged_data$scientificName <- trimws(gsub("^([A-Za-z]+)\\s{1,2}[Ss]?p{1,2}\\s?\\.$", '\\1', merged_data$scientificName))
+
+# Cleaning 4
+# Check for genus and sp. without space
+grep("^[A-Za-z]+sp\\.$", merged_data$scientificName, perl=TRUE, value=TRUE)
+# Remove sp.
+merged_data$scientificName <- trimws(gsub("^([A-Za-z]+)sp\\.$", '\\1', merged_data$scientificName))
+
+# Check for anything ended with p. and . (dot)
+grep("^[A-Za-z ]+p\\.$", merged_data$scientificName, perl=TRUE, value=TRUE)
+grep("^[A-Za-z ]+\\.$", merged_data$scientificName, perl=TRUE, value=TRUE)
+
+# Cleaning 5 and 6
+# Check "Ada 3 kata, kata ke-3 huruf besar" on scientificName field
+grep("^[A-Za-z]+\\s[a-z]+\\s[A-Z](?:[a-z]+)?$", merged_data$scientificName, perl=TRUE, value=TRUE)
+grep("^[A-Za-z]+\\s[a-z]+\\s[A-Z]\\.[A-Z]?(?:.[A-Z][a-z])?(?:[a-z])?$", merged_data$scientificName, perl=TRUE, value=TRUE)
+# Remove all with capital letter on the third word
+merged_data$scientificName <- trimws(gsub("^([A-Za-z]+\\s[a-z]+)\\s[A-Z](?:[a-z]+)?$", '\\1', merged_data$scientificName))
+merged_data$scientificName <- trimws(gsub("^([A-Za-z]+\\s[a-z]+)\\s[A-Z]\\.[A-Z]?(?:.[A-Z][a-z])?(?:[a-z])?$", '\\1', merged_data$scientificName))
+
+# Cleaning 7
+# Check anything with number on scientificName field > make a variable
+sp_number <- grep("^[A-Za-z]+\\s?[Ss]?p?\\.?\\s?\\.?[A-Za-z]?\\d\\.?$", merged_data$scientificName, perl=TRUE)
+# Copy anything with number on scientificName field to remarks field
+merged_data$remarks[sp_number] <- merged_data$scientificName[sp_number]
+# Check unidentified species
+grep("^[A-Za-z]+\\s\\d$", merged_data$scientificName, perl=TRUE, value = TRUE)
+# Check anything with number on scientificName field
+grep("^[A-Za-z]+\\s?[Ss]?p?\\.?\\s?\\.?[A-Za-z]?\\d\\.?$", merged_data$scientificName, perl=TRUE, value = TRUE)
+# Remove number and words with number on scientificName field
+merged_data$scientificName <- trimws(gsub("^([A-Za-z]+)\\s?[Ss]?p?\\.?\\s?\\.?[A-Za-z]?\\d\\.?$", '\\1', merged_data$scientificName))
+
+# Cleaning 8
+# Recode unidentified species to "Plantae"
+merged_data$scientificName <- car::recode(merged_data$scientificName, "c('Morfospesies', 'Rumput', 'Paku')= 'Plantae'")
+
+# Cleaning 11
+# Remove -- in the end
+merged_data$scientificName <- trimws(gsub("^([A-Za-z ]+)--$", '\\1', merged_data$scientificName))
+
+# Cleaning 12
+merged_data$scientificName <- trimws(gsub("^([A-Za-z]+\\s)cf(\\s[A-Za-z]+)$", '\\1cf.\\2', merged_data$scientificName))
+cf.with.uppercase.species <- function(value = FALSE) {
+  return(grep("^[A-Za-z]+\\scf\\.?\\s[A-Z][a-z]+$", merged_data$scientificName, perl=TRUE, value=value))
+}
+merged_data$scientificName[cf.with.uppercase.species()] <- firstup(tolower((cf.with.uppercase.species(TRUE))))
+
+# Cleaning 13
+# TODO: here
```

Comment:
case 13 ini memang belum kah? list of identified typos nya ada di comment di atas, hehe